test/src/test/java/org/apache/rocketmq/test/smoke/NormalMessageSendAndRecvIT.java

```diff
@@ -57,13 +60,33 @@ public void setUp() throws Exception {
 
     @After
     public void tearDown() {
-        super.shutdown();
+        BaseConf.shutdown();
     }
 
     @Test
     public void testSynSendMessage() throws Exception {
+        List<MessageQueue> messageQueueList;
+        ConsumeStats consumeStats;
+
+        // Wait till producer and consumer are fully started.
+        while (true) {
+            try {
+                consumeStats = defaultMQAdminExt.examineConsumeStats(group);
+                messageQueueList = producer.getProducer().fetchPublishMessageQueues(topic);
+                if (!messageQueueList.isEmpty() && null != consumeStats
+                    && consumeStats.getOffsetTable().keySet().containsAll(messageQueueList)) {
+                    break;
+                } else {
+
+                    TimeUnit.MILLISECONDS.sleep(100);
```

Comment:
Could we use awaitility to replace the sleep here?

Reply:
It's virtually the same busy polling way.